csharp/src/Ice/OutgoingResponseFrame.cs

```diff
@@ -107,92 +99,138 @@ public sealed class OutgoingResponseFrame
         /// <summary>Creates a new outgoing response frame with the given payload.</summary>
         /// <param name="request">The incoming request for which this constructor creates a response.</param>
         /// <param name="payload">The payload for this response frame.</param>
-        // TODO: add parameter such as "bool assumeOwnership" once we add memory pooling.
-        // TODO: should we pass the payload as a list of segments, or maybe add a separate
-        // ctor that accepts a list of segments instead of a single segment
         public OutgoingResponseFrame(IncomingRequestFrame request, ArraySegment<byte> payload)
-            : this(request)
+            : this(request.Protocol, request.Encoding, request.CompressionStatus)
         {
-            if (payload[0] == (byte)ReplyStatus.OK || payload[0] == (byte)ReplyStatus.UserException)
+            // Always reply status byte or result type byte followed by encapsulation.
```

Comment:
Did you mean "Always encode the reply status or result type byte followed by the encapsulation"?

Reply:
Reworked this code and comments.